src/NHibernate/Hql/Ast/ANTLR/QueryTranslatorImpl.cs

```diff
@@ -48,12 +49,14 @@ public partial class QueryTranslatorImpl : IFilterTranslator
 		/// <param name="parsedQuery">The hql query to translate</param>
 		/// <param name="enabledFilters">Currently enabled filters</param>
 		/// <param name="factory">The session factory constructing this translator instance.</param>
+		/// <param name="queryLoaderFactory">A query loader factory method.</param>
 		public QueryTranslatorImpl(
 			string queryIdentifier,
 			IASTNode parsedQuery,
 			IDictionary<string, IFilter> enabledFilters,
-			ISessionFactoryImplementor factory)
-			: this(queryIdentifier, parsedQuery, enabledFilters, factory, null)
+			ISessionFactoryImplementor factory,
+			Func<QueryTranslatorImpl, ISessionFactoryImplementor, SelectClause, IQueryLoader> queryLoaderFactory)
```

Comment:
It is not a pattern in-use in NHibernate. It is hard to supply by configuration a factory which is just a method pointer, while we can easily do this for a factory interface. So, I think you should instead use something like:

```suggestion
			IQueryLoaderFactory queryLoaderFactory)
```

with:
```
/// <summary>
/// Creates query loaders.
/// <summary>
public interface IQueryLoaderFactory
{
    /// <summary>
    /// Creates a query loader.
    /// <summary>
    /// ...
    IQueryLoader Create(QueryTranslatorImpl queryTranslator, ISessionFactoryImplementor factory, SelectClause selectClause); 
}

Reply:
Nice solution.